funding/manager.go

```diff
@@ -1388,10 +1388,29 @@ func (f *Manager) handleFundingOpen(peer lnpeer.Peer,
 	}
 	reservation.SetOurUpfrontShutdown(shutdown)
 
+	// Pack it into the ExtraData of the message as a TLV record.
+	var extraData lnwire.ExtraOpaqueData
```

Comment:
Hmm I see how this jumps forward to make the fields "TLV" native, but I think it may be better to hide more or this information from the caller, the way it was in the prior diff. So just setting the actual _fields_ on the struct, and letting the `Encode/Decode` methods do the rest.

As now we've fored the funding manager to keep up with all the future TLV types vs hiding the details behind `Encode/Decode` and using `nil`ness to indicate if a field was set or not.